fixit/common/unused_suppressions.py

```diff
@@ -0,0 +1,285 @@
+                        node,
+                        attribute_name,
+                        idx,
+                        local_supp_comment,
+                        comment_physical_line,
+                    )
+
+    def _handle_node_with_leading_lines(self, node: cst.CSTNode) -> None:
+        if hasattr(node, "leading_lines") and getattr(node, "leading_lines"):
+            self._handle_emptyline_sequence_attribute(node, "leading_lines")
+
+    def visit_Module(self, node: cst.Module) -> None:
+        if node.header:
+            self._handle_emptyline_sequence_attribute(node, "header")
+        if node.footer:
+            self._handle_emptyline_sequence_attribute(node, "footer")
+
+    def visit_SimpleStatementLine(self, node: cst.SimpleStatementLine) -> None:
+        self._handle_node_with_leading_lines(node)
+
+    def visit_If(self, node: cst.If) -> None:
+        self._handle_node_with_leading_lines(node)
+
+    def visit_Else(self, node: cst.Else) -> None:
+        self._handle_node_with_leading_lines(node)
+
+    def visit_BaseCompoundStatement(self, node: cst.BaseCompoundStatement) -> None:
+        self._handle_node_with_leading_lines(node)
+
+    def visit_ExceptHandler(self, node: cst.ExceptHandler) -> None:
+        self._handle_node_with_leading_lines(node)
+
+    def visit_Finally(self, node: cst.Finally) -> None:
+        self._handle_node_with_leading_lines(node)
+
+    def visit_Try(self, node: cst.Try) -> None:
+        self._handle_node_with_leading_lines(node)
+
+    def visit_Decorator(self, node: cst.Decorator) -> None:
+        self._handle_node_with_leading_lines(node)
+
+    def visit_FunctionDef(self, node: cst.FunctionDef) -> None:
+        if node.leading_lines:
+            self._handle_emptyline_sequence_attribute(node, "leading_lines")
+        if node.lines_after_decorators:
+            self._handle_emptyline_sequence_attribute(node, "lines_after_decorators")
+
+    def visit_ClassDef(self, node: cst.ClassDef) -> None:
+        if node.leading_lines:
+            self._handle_emptyline_sequence_attribute(node, "leading_lines")
+        if node.lines_after_decorators:
+            self._handle_emptyline_sequence_attribute(node, "lines_after_decorators")
+
+    def visit_With(self, node: cst.With) -> None:
+        self._handle_node_with_leading_lines(node)
+
+    def visit_For(self, node: cst.For) -> None:
+        self._handle_node_with_leading_lines(node)
+
+    def visit_While(self, node: cst.While) -> None:
+        self._handle_node_with_leading_lines(node)
+
+    def visit_ParenthesizedWhitespace(self, node: cst.ParenthesizedWhitespace) -> None:
+        if node.empty_lines:
+            self._handle_emptyline_sequence_attribute(node, "empty_lines")
+
+    def _handle_suppression_comment(
+        self,
+        parent_node: cst.CSTNode,
+        parent_attribute_name: str,
+        index: int,
+        local_supp_comment: SuppressionComment,
+        comment_physical_line: int,
+    ) -> None:
```

Comment:
This looks like something can be refactored to be generic for reuse in LibCST in the future.